File: sample/src/main/java/com/rjsvieira/recyclerviewmultipleviewtypesadapter/sample/adapter/SampleListAdapter.java
package com.rjsvieira.recyclerviewmultipleviewtypesadapter.sample.adapter;

import com.rjsvieira.recyclerviewmultipleviewtypesadapter.ListBindAdapter;
import com.rjsvieira.recyclerviewmultipleviewtypesadapter.sample.SampleData;
import com.rjsvieira.recyclerviewmultipleviewtypesadapter.sample.binder.Sample1Binder;
import com.rjsvieira.recyclerviewmultipleviewtypesadapter.sample.binder.Sample2Binder;
import com.rjsvieira.recyclerviewmultipleviewtypesadapter.sample.binder.Sample3Binder;

import java.util.List;

public class SampleListAdapter extends ListBindAdapter {

    public SampleListAdapter() {
        addAllBinder(new Sample1Binder(this),
                new Sample2Binder(this),
                new Sample3Binder(this));
    }

    public void setSample2Data(List<SampleData> dataSet) {
        ((Sample2Binder) getDataBinder(1)).addAll(dataSet);
    }

}

File: sample/src/main/java/com/rjsvieira/recyclerviewmultipleviewtypesadapter/sample/adapter/SampleListAdapter.kt
package com.rjsvieira.recyclerviewmultipleviewtypesadapter.sample.adapter

import com.rjsvieira.recyclerviewmultipleviewtypesadapter.DataBindAdapter
import com.rjsvieira.recyclerviewmultipleviewtypesadapter.sample.SampleData
import com.rjsvieira.recyclerviewmultipleviewtypesadapter.sample.binder.Sample1Binder
import com.rjsvieira.recyclerviewmultipleviewtypesadapter.sample.binder.Sample2Binder
import com.rjsvieira.recyclerviewmultipleviewtypesadapter.sample.binder.Sample3Binder

class SampleListAdapter : DataBindAdapter() {

    init {
        addBinder(Sample1Binder(this))
        addBinder(Sample2Binder(this))
        addBinder(Sample3Binder(this))
    }

    fun setSample2Data(dataSet: List<SampleData>) {
        (getBinderAtPosition(1) as Sample2Binder).addAll(dataSet)
    }

}
